File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode prev = null;
        ListNode pointer = head;

        while(pointer != null){
            if(pointer.val == val){
                if(prev == null){
                    head = pointer.next;
                } 
                else {
                    prev.next = pointer.next;
                }
            } 
            else {
                prev = pointer;
            }
            pointer = pointer.next;
        }

        return head;
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* prev = NULL;
        ListNode* pointer = head;

        while(pointer != NULL){
            if(pointer->val == val){
                if(prev == NULL) head = pointer->next;
                else prev->next = pointer->next;
            }
            else prev = pointer;
            pointer = pointer->next;
        }

        return head;
    }
};